### ISSUE TITLE
IconLayer onError src fallback

### ISSUE THREAD

**Issue by marioaks** [visgl/deck.gl#4149]
Hey! 

I've been using the IconLayer auto packing feature, where I'm fetching each icon from a url. I, however, cannot be sure that the url I'm passing into the IconLayer is a valid url, so sometimes the layer is throwing the following error:

> Unhandled Rejection (Error): Could not load image blob:http://localhost:3000/53961fd6-826e-42f8-b590-9af0dc29fc1b: [object Event]. _src/lib/parsers/parse-to-html-image.js:21_

My question is, is there a way to overwrite the icon-manager and set an onerror fallback image src in case the url that I passed in fails? If I can't, what would be the best way of dealing with this?

Thanks!

**Currently using deck.gl version 8.0.6**

**Comment by Pessimistress** [comment]
#5103 adds an `onIconError` callback. With this feature you'll be able to do something like:

```js
const badUserIds = new Set();

function redraw() {
  const layer = new IconLayer({
    // ...
    getIcon: d => {
      return {
        url: badUserIds.has(d.userId) ? BROKEN_IMAGE : `https://avatars3.githubusercontent.com/u/${d.userId}?v=4`,
        width: 128,
        height: 128
      };
    },
    updateTriggers: {
      getIcon: badUserIds.size
    },
    onIconError: e => {
      badUserIds.add(e.source.userId);
      redraw();
    }
  });

  deck.setProps({layers: [layer]});
}
```

I can publish an alpha release once it's landed.

**Comment by Pessimistress** [comment]
`onIconError` callback is available in 8.4.0-alpha.2
